Add controllers

olegfxf opened the pull request.

olegfxf committed:
feat: Users

olegfxf committed:
feat: Users
fix: UserDto

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items
feat: full Items

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items
feat: full Items
fix: ItemService search

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items
feat: full Items
fix: ItemService search
feat: unit tests

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items
feat: full Items
fix: ItemService search
feat: unit tests+

olegfxf committed:
feat: Users
fix: UserDto
feat: 0.5 Items
feat: full Items
fix: ItemService search
feat: unit tests

olegfxf committed:
fix: codeStyle

olegfxf committed:
fix: codeStyle
fix: abstract Storage
fix: userService is abstract
fix: abstract Service

olegfxf committed:
fix: codeStyle
fix: abstract Storage
fix: userService is abstract
fix: abstract Service

olegfxf committed:
fix: codeStyle
fix: abstract Storage
fix: userService is abstract
fix: abstract Service

johnskyi commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,131 @@
 @RestController
 @RequestMapping("/items")
 public class ItemController {
+    public ItemService itemService;
```
private final

olegfxf replied:
Исправлено

johnskyi commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,131 @@
 @RequestMapping("/items")
 public class ItemController {
+    public ItemService itemService;
+    AbstractStorage<Item> itemStorage;
```
private final

olegfxf replied:
Исправлено

johnskyi commented on src/main/java/ru/practicum/shareit/item/ItemController.java:
```diff
@@ -1,12 +1,131 @@
+
+
+        try {
+            Item item = itemStorage.getById(id);
```
вся эта логика предпочтительно должна находится в сервисе. В контроллере только вызовы методов сервиса

olegfxf replied:
Вся бизнес-логика проекта находится в сервисе. В данном случае речь идет о валидации данных и, затем, их сохранении. Посмотрел несколько статей по применению @PatchMapping. Все содержат стандартные операции. В контроллере извлекают объект из Сервиса или Сториджа, патчат его и возвращают обратно через Сервис или Сторидж:
https://cassiomolin.com/2019/06/10/using-http-patch-in-spring/
https://www.baeldung.com/spring-rest-json-patch
https://stackoverflow.com/questions/70216801/spring-partial-bean-validation-in-patch-request

olegfxf committed:
fix: codeStyle
fix: abstract Storage
fix: userService is abstract
fix: abstract Service
fix: notes
